lib/options.js

```diff
@@ -87,9 +87,16 @@ export function normalizeOptions(options) {
     const sourceType = normalizeSourceType(options.sourceType);
     const ranges = options.range === true;
     const locations = options.loc === true;
+    const allowReserved = ecmaVersion === 3 ? "never" : null;
```

Comment:
Acorn does set [`null`](https://github.com/acornjs/acorn/blob/41068c92ecba93b01f0193e976f893b15d07ea20/acorn/src/options.js#L32) as the initial value and checks only if it is [`true`](https://github.com/acornjs/acorn/blob/49ac0492013fd201ebc3c0775631743f75111914/acorn/src/state.js#L14) or [`"never"`](https://github.com/acornjs/acorn/blob/49ac0492013fd201ebc3c0775631743f75111914/acorn/src/expression.js#L332), so this works well.

However, [documentation](https://github.com/acornjs/acorn/tree/master/acorn#interface) for this option mentions only `true`, `false` and `"never"` as allowed values, so it might be safe to use `false` here instead of `null`, or just omit the property?

Reply:
just thought another point: could it be "never" for all ecma versions? (As of now, users cannot set the option)

Reply:
> just thought another point: could it be "never" for all ecma versions? (As of now, users cannot set the option)

It wouldn't be correct per the specification. `"never"` means that acorn disallows reserved words as property names, but as of [ECMAScript 5](https://262.ecma-international.org/5.1/) it's a valid code:

> _MemberExpression_ :
> &nbsp;&nbsp;&nbsp;&nbsp; _MemberExpression_. _IdentifierName_
> _PropertyName_ :
> &nbsp;&nbsp;&nbsp;&nbsp; _MemberExpression_. _IdentifierName_

[Identifier Names and Identifiers](https://262.ecma-international.org/5.1/#sec-7.6)

_Identifier_ cannot be a reserved word, but _IdentifierName_ can.

Reply:
👍 when ecma version > 5, it defaults to `false`(not `"never"`), so it's spec-compliant. we just need to set it when ecma version = 3.

Reply:
> However, documentation for this option mentions only true, false and "never" as allowed values, so it might be safe to use false here instead of null, or just omit the property?

I tried this initially but as @aladdin-add points out, `false` has different meaning based on `ecmaVersion` so other tests failed. Setting to `null` is the simplest approach that gets us where we want to be.

Reply:
Hmm, I just tried with `const allowReserved = ecmaVersion === 3 ? "never" : false;` and all tests are passing. It could be that we have different versions of acorn locally installed.

I have concerns that `null` might break without notice since it isn't an officially allowed value. What do you think about omitting the property? That would make no observable changes in the options object we're passing to acorn when `ecmaVersion` isn't 3.

Reply:
Hmm that’s possible, I’ll try it again.